File: a010(prime_factorization)/C++/prime_factorization.cpp
#include <iostream>
#include <cmath>

int main(void) {
  int number;
  while (std::cin >> number) {
    int count = 0;
    if (number % 2 == 0) {
      while (number % 2 == 0) {
        number /= 2;
        count++;
      }
      if (count != 1) std::cout << "2^" << count;
      else std::cout << "2";
      if (number != 1) std::cout << " * ";
      count = 0;
    }
    for (int i = 3; i < number + 1; i += 2) {
      if ((number % i) == 0) {
        while (number % i == 0) {
          number /= i;
          count++;
        }
        if (count != 1) std::cout << i << "^" << count;
        else std::cout << i;
        if (number != 1) std::cout << " * ";
        count = 0;
      }
    }
    std::cout << '\n';
  }

  return 0;
}

File: a010(prime_factorization)/Python/prime_factorization.py

while True:
    try:
        number = int(input())
        prime = []
        while number != 1:
            if number % 2 == 0:
                prime.append(2)
                number //= 2
                continue
            for i in range(3, number + 1, 2):
                if number % i == 0:
                    prime.append(i)
                    number //= i
                    break
        p, c, i = [], [], 0
        while i < len(prime):
            p.append(prime[i])
            c.append(prime.count(prime[i]))
            i += prime.count(prime[i])
        for i in range(len(p) - 1):
            if c[i] != 1:
                print(f'{p[i]}^{c[i]} *', end = ' ')
            else:
                print(f'{p[i]} *', end = ' ')
        if c[len(p) - 1] != 1:
            print(f'{p[len(p) - 1]}^{c[len(p) - 1]}')
        else:
            print(f'{p[len(p) - 1]}')
    except:
        break
